### PR TITLE
docs: Add 1.12.0 upgrade guide

### PR DESCRIPTION
**Pull request by olivermrbl**

### CODE REVIEW

**Comment by olivermrbl** [comment]
@shahednasser good point, I've added a note on the breaking change. Feel free to change it to make it fit for the documentation, in case it's needed 👍

File: docs/content/upgrade-guides/medusa-core/1-12-0.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    quantity?: number;
+   }[],
+   context: PriceSelectionContext
+  ): Promise<Map<string, PriceSelectionResult>>
+    // ...
+  }
+}
+```
+
+You can learn more in the [Price Selection Strategy](../../modules/price-lists/price-selection-strategy.md#calculatevariantprice-method) documentation.
```
Comment (shahednasser) [inline docs/content/upgrade-guides/medusa-core/1-12-0.md:100]: This assumes we merge #4208 as well

**Review by olivermrbl — COMMENTED** [review]
I like the addition of the "Change Price..." section. Though, I would revert the before and after to not be wrapped in a class. It should just be the method signatures, as it applies to both our default implementation as well as any custom-built ones

**Comment by olivermrbl** [comment]
@shahednasser - okay I merge this now? :pra

**Comment by shahednasser** [comment]
@olivermrbl yes all good 👍🏻

**Pull request merged**